staging/src/k8s.io/apimachinery/pkg/api/resource/amount.go

```diff
@@ -203,6 +203,44 @@ func (a *int64Amount) Sub(b int64Amount) bool {
 	return a.Add(int64Amount{value: -b.value, scale: b.scale})
 }
 
+// Mul multiplies the provided b to the current amount, or
+// returns false if overflow or underflow would result.
+func (a *int64Amount) Mul(b int64) bool {
+	switch {
+	case a.value == 0:
+		return true
+	case b == 0:
+		a.value = 0
+		a.scale = 0
+		return true
+	case a.scale == 0:
+		c, ok := int64Multiply(a.value, b)
+		if !ok {
+			return false
+		}
+		a.value = c
+	case a.scale > 0:
+		c, ok := int64Multiply(a.value, b)
```

Comment:
This is super weird, because we're mutating `a` but can return false. My expectation would be that `a` is not mutated if we return false.

Reply:
scratch that, it's mutated explicitly below.

Reply:
Thanks for the review.

Yes, `int64Multiply()` doesn't mutate `a`, and `a` is explicitly mutated below.